app/http.php

```diff
@@ -0,0 +1,366 @@
+
+use OpenRuntimes\Proxy\Health\Health;
+use OpenRuntimes\Proxy\Health\Node;
+use Swoole\Atomic;
+use Swoole\Coroutine\Http\Client;
+use Utopia\Logger\Log;
+use Utopia\Logger\Logger;
+use Swoole\Coroutine\Http\Server;
+use Swoole\Http\Request as SwooleRequest;
+use Swoole\Http\Response as SwooleResponse;
+use Swoole\Runtime;
+use Swoole\Table;
+use Swoole\Timer;
+use Utopia\App;
+use Utopia\Balancing\Algorithm\Random;
+use Utopia\Balancing\Algorithm\RoundRobin;
+use Utopia\Balancing\Balancing;
+use Utopia\Balancing\Option;
+use Utopia\CLI\Console;
+use Utopia\Logger\Adapter;
+use Utopia\Logger\Adapter\AppSignal;
+use Utopia\Logger\Adapter\LogOwl;
+use Utopia\Logger\Adapter\Raygun;
+use Utopia\Logger\Adapter\Sentry;
+use Utopia\Registry\Registry;
+use Utopia\Swoole\Request;
+use Utopia\Swoole\Response;
+
+Runtime::enableCoroutine(true, SWOOLE_HOOK_ALL);
+
+$register = new Registry();
+
+$register->set('executorStates', function () {
+    $executorsCount = \count(\explode(',', (string) App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTORS', '')));
+    $executorStates = new Table($executorsCount);
+    $executorStates->column('hostname', Swoole\Table::TYPE_STRING, 128); // Same as key of row
+    $executorStates->column('status', Swoole\Table::TYPE_STRING, 8); // 'online' or 'offline'
+    $executorStates->column('state', Swoole\Table::TYPE_STRING, 16384); // State as JSON
+    $executorStates->create();
+    return $executorStates;
+});
+
+// Only used if round-robin algo is used
+$register->set('roundRobinAtomic', fn () => new Atomic(0));
+
+function fetchExecutorsState(bool $forceShowError = false): void
+{
+    global $register;
+
+    $executorStates = $register->get('executorStates');
+
+    $executors = \explode(',', (string) App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTORS', ''));
+
+    $health = new Health();
+
+    foreach ($executors as $executor) {
+        $health->addNode(new Node($executor));
+    }
+
+    $nodes = $health
+        ->run()
+        ->getNodes();
+
+    foreach ($nodes as $node) {
+        $status = $node->isOnline() ? 'online' : 'offline';
+        $hostname = $node->getHostname();
+
+        $oldStatus = $executorStates->exists($hostname) ? $executorStates->get($hostname)['status'] ?? null : null;
+        if ($forceShowError === true || ($oldStatus !== null && $oldStatus !== $status)) {
+            Console::success('Executor "' . $node->getHostname() . '" went ' . $status . '.');
+        }
+
+        $executorStates->set($node->getHostname(), [
+            'status' => $status,
+            'hostname' => $hostname,
+            'state' => \json_encode($node->getState())
+        ]);
+    }
+}
+
+/**
+ * Create logger instance
+ */
+$providerName = App::getEnv('OPEN_RUNTIMES_PROXY_LOGGING_PROVIDER', '');
+$providerConfig = App::getEnv('OPEN_RUNTIMES_PROXY_LOGGING_CONFIG', '');
+$logger = null;
+
+if (!empty($providerName) && !empty($providerConfig) && Logger::hasProvider($providerName)) {
```

Comment:
The way we init the logger instance globally and then pass it to logError function is a bit messy. We should also avoid using the global scope as much as we can. Best way to implement it, would be to return a logger instance for App::setResource.

Reply:
Moved to resource.